hasktorch/test/doctests.hs

```diff
@@ -4,12 +4,10 @@ import Test.DocTest
 
 main :: IO ()
 main = doctest
-  [ "-XOverloadedStrings"
-  , "-XOverloadedLists"
```

Comment:
The extensions listed here will apply to _all_ Haskell files in this repo, so you have to be careful which extensions you list here.

In general, it is not safe to list extensions like `OverloadedStrings` and `OverloadedLists`, which will cause type-checking unification problems when used on .hs files that don't expect them.

The safe solution to this is to use the Doctest `$setup` blocks in the Haskell source files, as I have done in `hasktorch/src/Torch/Typed/Functional.hs` above.  This turns on the extensions for _only_ the doctests.

Reply:
thank you, that is a great suggestion!